pyquil/paulis.py

```diff
@@ -696,6 +696,26 @@ def exp_wrap(param):
     return exp_wrap
 
 
+def exponentiate_commuting_pauli_sum(pauli_sum):
+    """
+    Returns a function that maps all substituent PauliTerms and sums them into a program. NOTE: Use
+    this function with care. Substituent PauliTerms should commute.
+
+    :param PauliSum pauli_sum: PauliSum to exponentiate.
+    :returns: A function that parametrizes the exponential.
+    :rtype: function
+    """
+    if not isinstance(pauli_sum, PauliSum):
+        raise TypeError("Argument 'pauli_sum' must be a PauliSum.")
+
+    fns = [exponential_map(term) for term in pauli_sum]
+
+    def combined_exp_wrap(param):
+        return sum([f(param) for f in fns], Program())
```

Comment:
IMO the syntax Program([f(param) for f in fns]) is clearer than adding the second argument to sum but it's not a big deal

Reply:
I agree, changed.

Reply:
oh yeah that is better. Maybe we should use that in some of the other code snippets too.